src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -85,40 +85,60 @@ static PSVersionInfo()
             //
             // The strings can be one of the following format examples:
             //    when powershell is built from a commit:
-            //      productVersion = '6.0.0-beta.7 Commits: 29 SHA: 52c6b...' convert to GitCommitId = 'v6.0.0-beta.7-29-g52c6b...'
-            //                                                                           PSVersion   = '6.0.0-beta.7'
+            //      productVersion = '7.3.0-preview.7 Commits: 29 SHA: 52c6b...' convert to GitCommitId = 'v7.3.0-preview.7-29-g52c6b...'
+            //                                                                           PSVersion   = '7.3.0-preview.7'
             //    when powershell is built from a release tag:
-            //      productVersion = '6.0.0-beta.7 SHA: f1ec9...'             convert to GitCommitId = 'v6.0.0-beta.7'
-            //                                                                           PSVersion   = '6.0.0-beta.7'
+            //      productVersion = '7.3.0-preview.7 SHA: f1ec9...'             convert to GitCommitId = 'v7.3.0-preview.7'
+            //                                                                           PSVersion   = '7.3.0-preview.7'
             //    when powershell is built from a release tag for RTM:
-            //      productVersion = '6.0.0 SHA: f1ec9...'                    convert to GitCommitId = 'v6.0.0'
-            //                                                                           PSVersion   = '6.0.0'
-            string rawGitCommitId;
-            string mainVersion = ProductVersion.Substring(0, ProductVersion.IndexOf(' '));
+            //      productVersion = '7.3.0 SHA: f1ec9...'                    convert to GitCommitId = 'v7.3.0'
+            //                                                                           PSVersion   = '7.3.0'
+            int index = ProductVersion.IndexOf(' ');
+            ReadOnlySpan<char> mainVersion = ProductVersion.AsSpan(0, index);
 
-            if (ProductVersion.Contains(" Commits: "))
-            {
-                rawGitCommitId = ProductVersion.Replace(" Commits: ", "-").Replace(" SHA: ", "-g");
-            }
-            else
-            {
-                rawGitCommitId = mainVersion;
-            }
+            string rawGitCommitId = ProductVersion.AsSpan(index).StartsWith(" Commits: ")
+                ? "v" + ProductVersion.Replace(" Commits: ", "-").Replace(" SHA: ", "-g")
+                : string.Concat("v".AsSpan(), mainVersion);
 
-            s_psSemVersion = new SemanticVersion(mainVersion);
```

Comment:
Avoid calling `SemanticVersion(string)`, which needs to parse the argument.
The PS version format is pre-defined, so we can greatly simplify the parsing.

Reply:
I have no strict objections. Only this is formally a breaking change. Besides, my intention was that world community migrate to semantic version.

If we will use SG we can leave these version as-is since with SG we can put constants in separate helper and avoid initialization of PSVersion class at startup at all.

Reply:
> WSManStackVersion 

Is this used at startup?
If yes, we could cache it in separate helper to avoid PSVersion init - it is my proposal for all that are used at startup.

Reply:
It is used for `$PSVersionTable.WSManStackVersion`. You cannot cache it, the check needs to happen on the user's machine, not where PowerShell gets built.

Reply:
Not used at startup, but the value for that key needs to come from the user's registry, not the machine that PowerShell was built on, so source generator doesn't help in this case.

Today, the Hashtable is initialized when `PSVerseionInfo` is accessed, so we pay the penalty even though the key itself is not used at startup. To eliminate that, we could have a delay initialized Hashtable, as what I said in https://github.com/PowerShell/PowerShell/pull/18210#discussion_r988120904